### ISSUE TITLE
Install action with enabled Atomic is returning success when there is ImagePullBackOff

### ISSUE THREAD

**Issue by IoanaIvanova** [helm/helm#9415]
Hi,

I've just observed the fact that [action install](https://github.com/helm/helm/blob/master/pkg/action/install.go#L173) called from whiting a golang client with configured Atomic=true results in no failure even if a pod is being in an ImagePullBackOff state.

From what I have checked the Atomic flag  activated the Wait flag too. So I suppose that when I enable those flags if my pod is in ImagePullBackOff state the install action should return with failure or error not success. Also the release that is returned as a result to the call of Run function has status deployed. 

I have saw in [this issue](https://github.com/helm/helm/issues/3296) that a similar behaviour was observed and the issue states that it is fixed.

Could you please tell me if I am missing something that I should configure in the install action or is it a regression?

Best Regards, 
Yoana

**Comment by mattfarina** [comment]
Waiting is a bit complicated because there is no standard in Kubernetes for waiting. It depends on the resource you deploy. The one described in your charts template. For example, if you have a Deployment Helm will look at the ReplicaSet it created and the number Kubernetes says is ready compared to max unavailable settings (you can see the code [here](https://github.com/helm/helm/blob/3e31983a37b08486f842bf6b04dd2f72628ddf37/pkg/kube/wait.go#L241-L248)). You'll notice it's not looking at the status of each pod because one pod not being up may not be an issue given the requested state. Something like a DaemonSet is handled differently.

Can you share what Kubernetes resources you are using wait on? The templates generated by Helm specifically.

**Comment by IoanaIvanova** [comment]
Hi, 

Thanks for the explanation. 

My Pod is Wrapped in a Job not Deployment. 
I saw the flag waitForJobsEnabled but as far as I understood it if I enable this flag it will wait until the Job is Completed but that is not the desired behaviour. The desired behaviour is to wait until the Job is in Running state.

From what I saw in the jobReady function the desired behaviour is not possible with the current implementation of wait for job resources. I think that the only way to fulfil the desired behaviour is to make a custom implementation which checks the Job status and when it detects that it is in Running state to return with success. Have I understood it right or is there a simpler way to fulfil the desired behaviour?

Best regards,
Yoana
